src/BuildScriptGenerator/Helpers/ListBlobsHelper.cs

```diff
@@ -2,22 +2,41 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license.
 // --------------------------------------------------------------------------------------------
+using System;
 using System.Linq;
+using System.Net.Http;
 using System.Xml.Linq;
 using Microsoft.Oryx.BuildScriptGenerator.Common;
 
 namespace Microsoft.Oryx.BuildScriptGenerator
 {
     internal static class ListBlobsHelper
     {
-        public static XDocument GetAllBlobs(string sdkStorageBaseUrl, string platform, System.Net.Http.HttpClient httpClient, string oryxSdkStorageAccountAccessToken)
+        public static XDocument GetAllBlobs(string sdkStorageBaseUrl, string platform, HttpClient httpClient, string oryxSdkStorageAccountAccessToken)
         {
             var oryxSdkStorageAccountAccessArgs = oryxSdkStorageAccountAccessToken?.TrimStart(new char[] { '?' }) ?? string.Empty;
 
             var url = string.Format(SdkStorageConstants.ContainerMetadataUrlFormat, sdkStorageBaseUrl, platform, string.Empty, oryxSdkStorageAccountAccessArgs);
-            var blobList = httpClient
-                .GetStringAsync(url)
-                .Result;
+            string blobList;
+            try
+            {
+                blobList = httpClient
+                    .GetStringAsync(url)
+                    .Result;
+            }
+
+            // AggregateException is thrown by .Result
+            catch (AggregateException ae)
+            {
+                throw new AggregateException(
+                        $"Http request to '{sdkStorageBaseUrl}' failed. {Constants.NetworkConfigurationHelpText}\n{ae}");
```

Comment:
Nit: please consider adding a bit more details to the error message, e.g.:
```
Http request to get the SDKs available to download (URL '{sdkStorageBaseUrl}') failed. ...
```

This might be more satisfying for the customers/customer support folks, as well as giving us a 100% certain way to differentiate between the error thrown here and the one with same text thrown in SdkStorageVersionProviderBase.cs.
Obviously I'd have the same suggestion there as well.